src/redux/book/bookSlice.js

```diff
@@ -1,51 +1,146 @@
-    state.books.push(action.payload);
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    add: (state, action) => {
+      state.books.push(action.payload);
+    },
+    remove: (state, action) => {
+      const id = action.payload;
+      const newState = state.books.filter((i) => i.item_id !== id);
+      state.books = newState;
+    },
   },
-  remove: (state, action) => {
-    const id = action.payload;
-    const newState = state.books.filter((i) => i.item_id !== id);
-    state.books = newState;
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchBooks.pending, (state) => {
+        state.isLoading = true;
+        state.isError = false;
+        state.error = null;
+      })
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        const data = convertData(action.payload);
+        state.books = data;
+        state.isSuccess = true;
+        state.isLoading = false;
+        state.isError = false;
+        state.error = null;
+      })
+      .addCase(fetchBooks.rejected, (state, action) => {
+        state.isSuccess = false;
+        state.isLoading = false;
+        state.isError = true;
+        state.error = action.payload;
+      })
+      .addCase(addBook.pending, (state) => {
+        state.adding = true;
+        state.addedError = null;
+        state.addedSuccess = false;
+      })
+      .addCase(addBook.fulfilled, (state, action) => {
+        state.books = [...state.books, action.payload];
+        state.adding = false;
+        state.addedError = null;
+        state.addedSuccess = true;
+      })
+      .addCase(addBook.rejected, (state, action) => {
+        state.adding = false;
+        state.addedError = action.payload;
+        state.addedSuccess = false;
+      })
+      .addCase(removeBook.pending, (state) => {
+        state.deleting = true;
+        state.deletedSuccess = false;
+        state.deletedError = null;
+      })
+      .addCase(removeBook.fulfilled, (state, action) => {
+        state.books = state.books.filter((i) => i.id !== action.payload);
+        state.deleting = false;
+        state.deletedSuccess = true;
+        state.deletedError = null;
+      })
+      .addCase(removeBook.rejected, (state, action) => {
+        state.deleting = false;
+        state.deletedSuccess = false;
+        state.deletedError = action.payload;
+      });
   },
 });
 
-// Actions
-export const AddBook = (book) => (dispatch) => {
-  dispatch({
-    type: 'add',
-    payload: book,
-  });
-};
+// Reducers
+export const bookReducer = bookSlice.reducer;
+// // Actions
+// export const AddBook = (book) => (dispatch) => {
+//   dispatch({
+//     type: 'add',
+//     payload: book,
+//   });
+// };
 
-export const RemoveBook = (id) => (dispatch) => {
-  dispatch({
-    type: 'remove',
-    payload: id,
-  });
-};
+// export const RemoveBook = (id) => (dispatch) => {
+//   dispatch({
+//     type: 'remove',
+//     payload: id,
+//   });
+// };
```

Comment:
- [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

Reply:
Thank you very much for bringing that to be noticed!🤓